uavcan/equipment/ice/reciprocating/1120.Status.uavcan

```diff
@@ -0,0 +1,178 @@
+#
+# Generic status message of a piston engine control system.
+#
+# All integer fields are required unless stated otherwise.
+# All floating point fields are optional unless stated otherwise; unknown/unapplicable fields should be set to NaN.
+#
+
+#
+# Abstract engine state. The flags defined below can provide further elaboration.
+# This is a required field.
+#
+uint2 state
+#
+# The engine is not running. This is the default state.
+# Next states: STARTING, FAULT
+#
+uint2 STATE_STOPPED = 0
+#
+# The engine is starting. This is a transient state.
+# Next states: STOPPED, RUNNING, FAULT
+#
+uint2 STATE_STARTING = 1
+#
+# The engine is running normally.
+# Some error flags may be set to indicate non-fatal issues, e.g. overheating.
+# Next states: STOPPED, FAULT
+#
+uint2 STATE_RUNNING = 2
+#
+# The engine can no longer function.
+# The error flags may contain additional information about the nature of the fault.
+# Next states: STOPPED.
+#
+uint2 STATE_FAULT = 3
+
+#
+# General status flags.
+# Note that not all flags are required. Those that aren't are prepended with a validity flag, which is, obviously,
+# always required; when the validity flag is set, it is assumed that the relevant flags are set correctly.
+# If the validity flag is cleared, then the state of the relevant flags should be ignored.
+# All unused bits must be cleared.
+#
+uint22 flags
+#
+# General error. This flag is required, and it can be used to indicate an error condition
+# that does not fit any of the other flags.
+# Note that the vendor may also report additional status information via the vendor specific status code
+# field of the NodeStatus message.
+#
+uint22 FLAG_GENERAL_ERROR                       = 1
+#
+# Error of the crankshaft sensor. This flag is optional.
+#
+uint22 FLAG_CRANKSHAFT_SENSOR_ERROR_SUPPORTED   = 2
+uint22 FLAG_CRANKSHAFT_SENSOR_ERROR             = 4
+#
+# Temperature levels. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_TEMPERATURE_SUPPORTED               = 8
+uint22 FLAG_TEMPERATURE_BELOW_NOMINAL           = 16    # Under-temperature warning
+uint22 FLAG_TEMPERATURE_ABOVE_NOMINAL           = 32    # Over-temperature warning
+uint22 FLAG_TEMPERATURE_OVERHEATING             = 64    # Critical overheating
+uint22 FLAG_TEMPERATURE_EGT_ABOVE_NOMINAL       = 128    # Exhaust gas over-temperature warning 
+#
+# Fuel pressure. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_FUEL_PRESSURE_SUPPORTED             = 256
+uint22 FLAG_FUEL_PRESSURE_BELOW_NOMINAL         = 512   # Under-pressure warning
+uint22 FLAG_FUEL_PRESSURE_ABOVE_NOMINAL         = 1024  # Over-pressure warning
+#
+# Detonation and misfire. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_DETONATION_SUPPORTED                = 2048
```

Comment:
That's tightly coupled with detonation, so I left it in one group.
Will add one more support flag